### PR TITLE
GH-43270: [Release] Fix input variables on post-01-tag.sh

### PR DESCRIPTION
**Pull request by raulcd**
### Rationale for this change

Scripts is failing to be executed

### What changes are included in this PR?

Fixing the script

### Are these changes tested?

Locally during the release

### Are there any user-facing changes?

No
* GitHub Issue: #43270

### CODE REVIEW

File: dev/release/post-01-tag.sh [hunk 1]
```diff
@@ -25,8 +25,11 @@ if [ "$#" -ne 2 ]; then
   exit
 fi
 
+version=$1
+rc=$2
+
 # Create the release tag and trigger the Publish Release workflow.
-release_candidate_tag=apache-arrow-${version}-rc${num}
+release_candidate_tag=apache-arrow-${version}-rc${rc}
 release_tag=apache-arrow-${version}
```
Comment (kou) [inline dev/release/post-01-tag.sh:33]:
```suggestion
release_tag=apache-arrow-${version}
release_candidate_tag=${release_tag}-rc${rc}
```

may be better.

**Review by kou — APPROVED** [review]
+1

We may want to add `set -u` to detect this type of problem.

**Comment by sgilmore10** [comment]
+ 1

Thanks for fixing this! Apologies I did not catch it in the original PR.

**Comment by assignUser** [comment]
We should probably have shellcheck running in the linter job...

**Pull request merged**